Accept label value as regexp (#171)
prometheus-community/prom-label-proxy: README.md

`prom-label-proxy` will enforce the `tenant=~"prometheus|alertmanager"` label selector in all requests.

To error out when the query already contains a label matcher that differs from the one the proxy would inject, you can use the `-error-on-replace` option. For example:

```
prom-label-proxy \
   -header-name X-Namespace \
   -label namespace \
   -upstream http://demo.do.prometheus.io:9090 \
   -insecure-listen-address 127.0.0.1:8080 \
   -error-on-replace
```

> You can match label value  using a regular expression with the `-regex-match` option. For example:

Author: Done

`prom-label-proxy` will enforce the `tenant=~"prometheus|alertmanager"` label selector in all requests.

> You can match label value  using a regular expression with the `-regex-match` option. For example:

Reviewer: (nit)
```suggestion
You can match the label value  using a regular expression with the `-regex-match` option. For example:
```

Author: Done.

```
prom-label-proxy \

>    -header-name X-Namespace \
>    -label namespace \

Reviewer: (suggestion) I'd use a static label value to make it more "visible".
```suggestion
   -label-value '^foo-.+$' \
   -label namespace \
```

Reviewer: I wonder if we shouldn't anchor the regexp by default (like Prometheus does for relabelings)? It would at least prevent too-permissive configurations.
It'd also be good to require that the compiled regex doesn't match the empty string.

Author: > I wonder if we shouldn't anchor the regexp by default (like Prometheus does for relabelings)? It would at least prevent too-permissive configurations.

Prometheus anchors already.

> It'd also be good to require that the compiled regex doesn't match the empty string.

Good idea.

   -upstream http://demo.do.prometheus.io:9090 \
   -insecure-listen-address 127.0.0.1:8080 \
   -regex-match
```

> > :warning: The above feature is experimental. Be careful when using this option, it can expose sensitive metrics by accident.

Reviewer: (nit) just to clarify that the onus is on the person configuring prom-label-proxy (e.g. it's not like prom-label-proxy would disclose data by itself).
```suggestion
> :warning: The above feature is experimental. Be careful when using this option, it may expose sensitive metrics if you use a too permissive expression.
```
